.github/workflows/gsprocessing_workflow.yml

```diff
@@ -0,0 +1,120 @@
+# Graphstorm CI for GSProcessing Only
+# Test execution order - [lint] -> [pytest]
+name: gsprocessing_workflow
+
+on:
+  push:
+    branches:
+      - main
+      - ci_dev
+    paths:
+      - '.github/workflow_scripts/gsprocessing.sh'
+      - 'graphstorm-processing/*'
+      - '.github/workflows/gsprocessing_workflow.yml'
+    workflow_dispatch:
+  pull_request_target:
+    paths:
+      - '.github/workflow_scripts/gsprocessing.sh'
+      - 'graphstorm-processing/*'
+      - '.github/workflows/gsprocessing_workflow.yml'
+
+permissions:
+  id-token: write
+  pull-requests: write
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+# All test scripts split in individual .sh files and moved to .github/workflow scripts
+env:
+  COMMAND-LINT: chmod +x ./lint_check.sh && ./lint_check.sh
+  COMMAND-GSProcessing: chmod +x ./gsprocessing.sh && ./gsprocessing.sh
+
+jobs:
+  lint_check:
```

Comment:
Here we are using the common `lint_check.sh` script right? That one does a bunch of checks on the main GS repo that would be unnecessary here (and vice versa), since we know (from our workflow definition) that only files under `./graphstom-processing` were modified.

I see two options to avoid this:
1. Create a new `gsprocessing-lint.sh` workflow script that only checks GSProcessing code and execute that. Then we remove the gsprocessing-related line from `lint_check.sh`.
2. Add an argument to `lint_check.sh` that tells it whether to run pylint on gsprocessing or graphstorm files. The GSProcessing `COMMAND_LINT` could then be `bash ./lint_check.sh gsprorcessing` and similarly, in the graphstorm workflow it would `bash ./lint_check graphstorm`.

I'd be OK with either option of the above.